### ISSUE TITLE
<td> tags with no <tr> are not properly handled

### ISSUE THREAD

**Issue by defunctzombie** [cheeriojs/cheerio#422]
I am being passed some broke-ass-html where there is a missing `<tr>` tag before a `<td>`. This causes my queryies for table rows to fail as cheerio (or maybe the htmlparser) is not scoping the `<td>` inside of a logical `<tr>` as the browser does.

``` js
var cheerio = require('cheerio');
var str = '<table><td>bar</td></tr></table>';
var $ = cheerio.load(str);
console.log($.html());
```

**Comment by rpedela** [comment]
+1

Are there any known workarounds? I have the exact same problem.

**Comment by rpedela** [comment]
I answered my own question. [jsdom](https://github.com/tmpvar/jsdom) does not have this problem as it just uses jQuery. The API is a bit ugly, but it works. If you use Node.js instead of io.js then use [jsdom 3.x.x](https://github.com/tmpvar/jsdom/tree/3.x). It might be worth looking at how jQuery handles this case.

```
npm install jsdom@3
```
```js
var jsdom = require('jsdom').jsdom;
var window = jsdom(body).parentWindow;
jsdom.jQueryify(window, 'http://code.jquery.com/jquery-2.1.1.js', function () {
    var $ = window.$;
    console.log($('body').html());
});
```

**Comment by defunctzombie** [comment]
What is the perf like for jsdom now? I remember that being a bit of a show
stopper for my needs when evaluating solutions.

**Comment by rpedela** [comment]
I don't know. It seems fine, but performance isn't that important for my use case. You can load a local version of jQuery to eliminate one HTTP call.

**Comment by fb55** [comment]
Fixed by #985
